compiler/import_map.rs

```diff
@@ -0,0 +1,2082 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+use crate::Result;
+
+use deno_core::ModuleSpecifier;
+use indexmap::IndexMap;
```

Comment:
`ImportMap` implementation hasn't really changed since it was introduced last year in May, maybe it would make more sense to make it a separate crate as well? It would still depend on `deno_core` for the sake of `ModuleSpecifier`, but could limit complexity by having a clear boundary between the crates. @kitsonk @ry what do you think?

Reply:
I think it will "move" here, it shouldn't be needed by the CLI anywhere else.  Though of course `compiler` is sort of becoming a bad name for this crate... but basically it is the all the higher order logic of dependency/handling/transforming/type checking of ES Modules under Deno, abstracted away from the fetching/caching/persistence/runtime execution considerations.